### PR TITLE
Enable windows ci

### LINKED ISSUE #805: Fix Windows CI caching and reenable Windows CI
If you could fix the caching for the Windows CI I would be eternally grateful. Otherwise I might have to disable since the network connection to the stackage infrastructure (`casa.fpcomplete.com`) has become very unreliable and stack builds fail all the time

_Originally posted by @pepeiborra in https://github.com/haskell/ghcide/pull/794#discussion_r488124365_

### PR DESCRIPTION
**Pull request by jneira**
* Closes #805

### CODE REVIEW

**Review by ndmitchell — APPROVED** [review]

**Review by pepeiborra — CHANGES_REQUESTED** [review]
This doesn't fix the Azure cache issue

**Comment by jneira** [comment]
It seems the workaround of packing the cache, copied from hie and needed due to azure issues with the cache task is no longer needed, so removing the pack/unpack is a win-win :smile: 
I think maybe we could remove it from linux jobs too, i'll do in a new pr.

File: .azure/windows-stack.yml [hunk 1]
```diff
@@ -47,19 +46,14 @@ jobs:
       stack install happy --stack-yaml $STACK_YAML
       stack install alex --stack-yaml $STACK_YAML
       stack install cabal-install --stack-yaml $STACK_YAML
-      # GHC 8.10 keeps crashing with various kinds of access violations and other
-      # errors so we retry 3 times.
+      # GHC 8.10.1 fails with ghc segfaults, using -fexternal-interpreter seems to make it working
+      # There are another transient errors like time outs downloading from stackage so we retry 3 times
```
Comment (pepeiborra) [inline .azure/windows-stack.yml:50]:
```suggestion
      # There are other transient errors like timeouts downloading from stackage so we retry 3 times
```

**Comment by pepeiborra** [comment]
Wow, it really seems to work. Nice!

**Review by pepeiborra — APPROVED** [review]

**Pull request merged**